// rules/lib/core.js
/**
 * Created by Shepard on 16/04/15.
 */
//regex

var perNumber_regex = /^\d{10}$/,
    userID_regex = /^[a-zA-Z]{1}([a-zA-Z0-9]|[._]){4,14}$/;


//get length with 3 randomly numeric and word
exports.getRandomAlphaNum = function (len, radix) {
    radix = radix ? 10 : 36;
    var rdmString = "";
    for (; rdmString.length < len; rdmString += Math.random().toString(radix).substr(2));
    return rdmString.substr(0, len);
};
//get unique id
exports.getUniqueID = function () {
    return Math.round(new Date().getTime() / 1000);
};

//get current time
exports.getCurrentTime = function () {
    return new Date().toISOString().slice(0, 19).replace('T', ' ');
};

//get response json file
exports.getResponseJson = function(successStatus,messageContent,statusCode){
    return {
        success:successStatus,
        message:messageContent,
        statusCode:statusCode
    };
};

//validation qr code with length
exports.getValidationOf_QR_code = function (str) {
    if (perNumber_regex.test(str)) {
        return true;
    }
    else {
        return false;
    }
};

//validation user id with length and other charactars
exports.getVlidationOd_UserID = function (str) {
    if (userID_regex.test(str)) {
        return true;
    }
    else {
        return false;
    }

};

//randomly get unique userID according to user name, with 10 lists
exports.getUserIDRandomlyAcoordingToUserName = function (userName) {
    var advisedUserID = [];
    for (var i = 0; i < 4; i++) {
        advisedUserID.push(userName+'_'+this.getRandomAlphaNum(3));

    };
    return advisedUserID;
};


//randomly gener the valiation code for password changing
exports.getRandomValidationCode = function () {
    return this.getRandomAlphaNum(5);
};
